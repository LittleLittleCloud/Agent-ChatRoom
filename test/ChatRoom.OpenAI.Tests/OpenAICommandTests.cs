using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using Spectre.Console.Cli;
using Spectre.Console.Testing;
using Xunit;

namespace ChatRoom.OpenAI.Tests;

public class OpenAICommandTests
{
    [Fact]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("ApprovalTests")]
    public async Task ItShowHelperTextTestAsync()
    {
        var app = new CommandAppTester();
        app.SetDefaultCommand<OpenAICommand>(OpenAICommand.Description);
        var result = await app.RunAsync("--help");

        Approvals.Verify(result.Output);
    }
}
