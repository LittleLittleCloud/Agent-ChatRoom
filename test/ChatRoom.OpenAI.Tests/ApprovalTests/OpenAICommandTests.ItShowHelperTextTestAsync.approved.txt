DESCRIPTION:
OpenAI agent for chat room.

The agent will use OpenAI API to generate response for the input message.

To use the agent, you need to provide a configuration file.
A configuration file is a json file with the following schema:
- https://raw.githubusercontent.com/LittleLittleCloud/Agent-ChatRoom/main/schema/chatroom_openai_configuration_schema.json

USAGE:
    testhost.dll [OPTIONS]

OPTIONS:
    -h, --help               Prints help information
    -v, --version            Prints version information
    -r, --room <ROOM>        The room name to create
    -p, --port <PORT>        The port to listen
    -c, --config <CONFIG>    Configuration file